libraries/AP_Scripting/applets/rev_thr_on_land.lua

```diff
@@ -0,0 +1,151 @@
+    - A good rangefinder (downward facing) capable of reliably reading anything between [0.1, 5]m
+    - A separate RC channel that will serve as an "arming" switch for reverse throttle operation. 
+      This script uses RC_ARM_CHANNEL_ for this purpose (i.e. reverse throttle is armed whenever this switch PWM >1500, 
+      set by RC_ARM_CHANNEL_HI_PWM_)
+    Params to adjust for a particular fw frame:
+    - THRESH_LAND_HEIGHT_ (m) - this is the height read by rangefinder which corresponds to a plane's height above ground
+      on touchdown. Set this a couple of cms higher than the height reading by your rangefider when the aircraft is on the ground
+      but well within the height that the rangefinder can reliably read.
+    - LAND_DET_TIME_ (s) - time during which rangefinder height estimate is consistently below THRESH_LAND_HEIGHT_
+    - MIN_GROUND_SPEED_ (m/s) - minimum ground speed at which the autopilot will disengage reverse thrust and switch to
+      idle forward thrust (FWD_THRUST_IDLE_PWM_)
+    - RC_REV_THROTTLE_CHANNEL_NUM_ - is the RC channel that tells autopilot to reverse motor directions (RCx_FUNCTION=64)
+    - REV_THRUST_SW_HIGH_PWM_ - this sets high position that the RCx_FUNCTION=64 switch should have for reverse throttle operation
+    - REV_THRUST_SW_LOW_PWM_ - this sets low position that the RCx_FUNCTION=64 switch should have for forward throttle operation
+    - REV_THRUST_HIGH_PWM_, REV_THRUST_IDLE_PWM_ - max and min pwm values for RC_CHANNEL_NUM_ when in reverse throttle mode
+    - FWD_THRUST_IDLE_PWM_ - min value of the motors when in forward throttle mode, set this to REV_THRUST_IDLE_PWM_ for smooth operation
+    - REV_THRUST_CHG_PWM_PS_(pwm) - this is the motors' linear acceleration time in seconds, from REV_THRUST_IDLE_PWM_ to REV_THRUST_HIGH_PWM_
+    - MAXTIME_REV_THRUST_APPLIED_MS_ (ms) - maximum time, in seconds, reverse throttle is allowed to apply
+    - TIME_THROT_WAIT_MS_ (ms) - time the throttle is suppressed, sitting at forward idle (used to ensure props are not spinning at a rate higher 
+      than FWD_THRUST_IDLE_PWM_ shortly before reverse throttle is applied)
+    - REV_THRUST_SPOOL_DPWM_ (pwm) - initial increase in pwm to apply when in reverse throttle mode
+]]--
+
+
+-- *****---PARAMS TO CHANGE---*****
+--REV arming parameters
+local RC_ARM_IN_CHANNEL_ = 6
+local ARM_CHANNEL_HI_PWM_ = 1500
+
+--REV switch setup
+local RC_REV_THROTTLE_CHANNEL_NUM_ = 11
+local REV_THRUST_SW_LOW_PWM_ = 1000
+local REV_THRUST_SW_HIGH_PWM_ = 2000
+
+--REV activation and spool up parameters
+local LAND_DET_TIME_ = 1
+local THRESH_LAND_HEIGHT_ = 0.12
+local TIME_THROT_WAIT_MS_ = 1000
+local REV_THRUST_SPOOL_DPWM_ = 60
+local REV_THRUST_CHG_PWM_PS_ = 150
+
+--REV strength and duration setup
+local FWD_THRUST_IDLE_PWM_ = 990
+local REV_THRUST_IDLE_PWM_ = 990
+local REV_THRUST_HIGH_PWM_ = 1600
+local MAXTIME_REV_THRUST_APPLIED_MS_ = 10000
+local MIN_GROUND_SPEED_ = -1
```

Comment:
You could add these parameters to a param table so users can set in the normal way. See: https://github.com/ArduPilot/ardupilot/blob/master/libraries/AP_Scripting/examples/param_add.lua

Reply:
Ok, will add the params and edit the PR.